hazelcast/src/main/java/com/hazelcast/sql/impl/expression/predicate/ComparisonPredicate.java

```diff
@@ -79,7 +90,17 @@ public Boolean eval(Row row, ExpressionEvalContext context) {
             return null;
         }
 
-        if (this.operand1.getType().getTypeFamily() == QueryDataTypeFamily.OBJECT) {
+        if (operand1.getType().getTypeFamily().equals(ROW)
```

Comment:
Can you extract `operand1.getType().getTypeFamily()` and `operand2.getType().getTypeFamily()` to variables? it will make code much clearer